src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1685 @@
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfOverreachingSelection1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (|]b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfOverreachingSelection2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (b)
+        |]{
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfOverreachingSelection3()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [|else if (b)
+        {|]
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionSelection1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if ([|b|])
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionSelection2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if [|(b)|]
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodySelection()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        [|{
+        }|]
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfConditionCaret()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if ([||]b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodyCaret1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        [||]{
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfBodyCaret2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)
+        {
+        }[||]
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnParentIf()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        [||]if (a)
```

Comment:
that surprises me actually.  why?

Reply:
The merging refactorings are currently always only offered on the inner/child/second if, as opposed to the outer/parent/first one. I went with the OP in #29347 mentioned they would like a merging refactoring on the inner if. Supporting both cases seems like it would add a nontrivial amount of complexity, and it also seems somewhat ambiguous to me, what happens if we can merge with both the parent and the child one? Which of these would the user expect to take over? Or do we offer both? This seemed like the nicest solution.